chain-extensions/unified-accounts/src/lib.rs

```diff
@@ -0,0 +1,84 @@
+// This file is part of Astar.
+
+// Copyright (C) 2019-2023 Stake Technologies Pte.Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+// Astar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Astar is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Astar. If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use astar_primitives::evm::{EvmAddress, UnifiedAddressMapper};
+use core::marker::PhantomData;
+use sp_runtime::DispatchError;
+
+use frame_support::{traits::Get, DefaultNoBound};
+use pallet_contracts::chain_extension::{
+    ChainExtension, Environment, Ext, InitState, Result as DispatchResult, RetVal,
+};
+use parity_scale_codec::Encode;
+pub use unified_accounts_chain_extension_types::Command::{self, *};
+
+#[derive(DefaultNoBound)]
+pub struct UnifiedAccountsExtension<T, UA>(PhantomData<(T, UA)>);
+
+impl<T, UA> ChainExtension<T> for UnifiedAccountsExtension<T, UA>
+where
+    T: pallet_contracts::Config + pallet_unified_accounts::Config,
+    UA: UnifiedAddressMapper<T::AccountId>,
+{
+    fn call<E>(&mut self, env: Environment<E, InitState>) -> DispatchResult<RetVal>
+    where
+        E: Ext<T = T>,
+    {
+        let mut env = env.buf_in_buf_out();
+        match env.func_id().try_into().map_err(|_| {
+            DispatchError::Other("Unsupported func id in Unified Accounts Chain Extension")
+        })? {
+            GetEvmAddress => {
+                let account_id: T::AccountId = env.read_as()?;
+
+                let base_weight = <T as frame_system::Config>::DbWeight::get().reads(1);
```

Comment:
It would be better to benchmark this in future.
It might look silly, but having an automated system that will detect changes, and update required weights is way better than hardcoding values.

Reply:
I can add the benchmarks for reading the mappings via `UnifiedAddressMapper` interface in pallet's existing benchmarks.
That we can use in other places also, like XVM automatic default mapping creation which currently charge 1 db read weight too.

I can create a issue and a followup PR later, don't want to make this PR more messy, if that is fine :)

Reply:
Agree!

Reply:
Followup issue - https://github.com/AstarNetwork/Astar/issues/1062